cmd/prysmctl/validator/proposer_settings.go

```diff
@@ -0,0 +1,112 @@
+	ctx, span := trace.StartSpan(c.Context, "prysmctl.getProposerSettings")
+	defer span.End()
+	if !c.IsSet(ValidatorHostFlag.Name) {
+		return fmt.Errorf("no --%s flag value was provided", ValidatorHostFlag.Name)
+	}
+	if !c.IsSet(TokenFlag.Name) {
+		return fmt.Errorf("no --%s flag value was provided", TokenFlag.Name)
+	}
+	defaultFeeRecipient := params.BeaconConfig().DefaultFeeRecipient.Hex()
+	if c.IsSet(ProposerSettingsOutputFlag.Name) {
+		if c.IsSet(DefaultFeeRecipientFlag.Name) {
+			recipient := c.String(DefaultFeeRecipientFlag.Name)
+			if err := validateIsExecutionAddress(recipient); err != nil {
+				return err
+			}
+			defaultFeeRecipient = recipient
+		} else {
+			promptText := "please enter a default fee recipient address (an ethereum address in hex format)"
+			resp, err := prompt.ValidatePrompt(r, promptText, validateIsExecutionAddress)
+			if err != nil {
+				return err
+			}
+			defaultFeeRecipient = resp
+		}
+	}
+
+	cl, err := validator.NewClient(c.String(ValidatorHostFlag.Name), client.WithAuthenticationToken(c.String(TokenFlag.Name)))
+	if err != nil {
+		return err
+	}
+	validators, err := cl.GetValidatorPubKeys(ctx)
+	if err != nil {
+		return err
+	}
+	feeRecipients, err := cl.GetFeeRecipientAddresses(ctx, validators)
+	if err != nil {
+		return err
+	}
+
+	log.Infoln("===============DISPLAYING CURRENT PROPOSER SETTINGS===============")
+
+	for index := range validators {
+		log.Infof("validator: %s. fee-recipient: %s", validators[index], feeRecipients[index])
+	}
+
+	if c.IsSet(ProposerSettingsOutputFlag.Name) {
+		log.Infof("the default fee recipient is set to %s", defaultFeeRecipient)
+		var builderSettings *validatorpb.BuilderConfig
+		if c.Bool(WithBuilderFlag.Name) {
+			builderSettings = &validatorpb.BuilderConfig{
+				Enabled:  true,
+				GasLimit: validatorType.Uint64(params.BeaconConfig().DefaultBuilderGasLimit),
+			}
+		} else {
+			log.Infof("default builder settings can be included with the `--%s` flag", WithBuilderFlag.Name)
+		}
+		proposerConfig := make(map[string]*validatorpb.ProposerOptionPayload)
+		for index, val := range validators {
+			proposerConfig[val] = &validatorpb.ProposerOptionPayload{
+				FeeRecipient: feeRecipients[index],
+				Builder:      builderSettings,
+			}
+		}
+		fileConfig := &validatorpb.ProposerSettingsPayload{
+			ProposerConfig: proposerConfig,
+			DefaultConfig: &validatorpb.ProposerOptionPayload{
+				FeeRecipient: defaultFeeRecipient,
+				Builder:      builderSettings,
+			},
+		}
+		b, err := json.Marshal(fileConfig)
+		if err != nil {
+			return err
+		}
+		if err := file.WriteFile(c.String(ProposerSettingsOutputFlag.Name), b); err != nil {
+			return err
+		}
+		log.Infof("successfully created `%s`. settings can be imported into validator client using --%s flag.", c.String(ProposerSettingsOutputFlag.Name), flags.ProposerSettingsFlag.Name)
+	}
+
+	return nil
+}
+
+func validateIsExecutionAddress(input string) error {
```

Comment:
This function is incorrect because `0xpppp...` with the correct length will pass. You should use `bytesutil.IsHex` + the length check. It would be nice to add some unit tests too.